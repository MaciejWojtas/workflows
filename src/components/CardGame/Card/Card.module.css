.card {
  @apply w-24 bg-white rounded-xl p-4 xl:p-6 text-4xl font-bold xl:rounded-3xl xl:text-7xl  xl:w-48;
}

.card-desc {
  @apply flex flex-col bg-white justify-start;
}

.card-desc__rank {
  @apply text-left;
}
.card-desc__rank--red {
  @apply text-love;
}

.card-desc__img {
  @apply w-5 h-5 m-1 xl:m-4 xl:h-9 xl:w-9;
}

.card-suit {
  @apply flex justify-end;
}

.card-suit__img {
  @apply w-12 h-12 xl:w-20  xl:h-20;
}
