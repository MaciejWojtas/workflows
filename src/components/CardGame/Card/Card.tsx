import { FunctionComponent } from 'react';
import { getSuitImage, isRedCard } from '../../../helpers/card';
import { Card as CardType } from '../../../types/card';
import styles from './Card.module.css';

type CardProps = {
  card: CardType;
  className?: string;
};

const Card: FunctionComponent<CardProps> = ({ card, className }) => {
  const Image = getSuitImage(card);

  return (
    <div className={`${styles.card} ${className || ''}`}>
      <div className={styles['card-desc']}>
        <span
          className={`${styles['card-desc__rank']} ${
            isRedCard(card) ? styles['card-desc__rank--red'] : ''
          }`}
        >
          {card.rank}
        </span>
        <Image className={styles['card-desc__img']} />
      </div>
      <div className={styles['card-suit']}>
        <Image className={styles['card-suit__img']} />
      </div>
    </div>
  );
};

export default Card;
