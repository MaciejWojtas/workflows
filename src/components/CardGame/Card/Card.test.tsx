import React from 'react';
import renderer from 'react-test-renderer';
import { cloverCard, diamondCard, heartCard, spadeCard } from '../../../tests/mocks/cards';
import Card from './Card';

it('renders correctly with Clover card', () => {
  const tree = renderer.create(<Card card={cloverCard} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with Heart card', () => {
  const tree = renderer.create(<Card card={heartCard} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with Spade card', () => {
  const tree = renderer.create(<Card card={spadeCard} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with Diamond card', () => {
  const tree = renderer.create(<Card card={diamondCard} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with Class Name', () => {
  const tree = renderer.create(<Card className="class-name" card={diamondCard} />).toJSON();

  expect(tree).toMatchSnapshot();
});
