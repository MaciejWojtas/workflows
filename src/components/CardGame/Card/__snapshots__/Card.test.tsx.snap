// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with Class Name 1`] = `
<div
  className="card class-name"
>
  <div
    className="card-desc"
  >
    <span
      className="card-desc__rank card-desc__rank--red"
    >
      A
    </span>
    <svg
      className="card-desc__img"
    >
      Diamond.svg
    </svg>
  </div>
  <div
    className="card-suit"
  >
    <svg
      className="card-suit__img"
    >
      Diamond.svg
    </svg>
  </div>
</div>
`;

exports[`renders correctly with Clover card 1`] = `
<div
  className="card "
>
  <div
    className="card-desc"
  >
    <span
      className="card-desc__rank "
    >
      A
    </span>
    <svg
      className="card-desc__img"
    >
      Clover.svg
    </svg>
  </div>
  <div
    className="card-suit"
  >
    <svg
      className="card-suit__img"
    >
      Clover.svg
    </svg>
  </div>
</div>
`;

exports[`renders correctly with Diamond card 1`] = `
<div
  className="card "
>
  <div
    className="card-desc"
  >
    <span
      className="card-desc__rank card-desc__rank--red"
    >
      A
    </span>
    <svg
      className="card-desc__img"
    >
      Diamond.svg
    </svg>
  </div>
  <div
    className="card-suit"
  >
    <svg
      className="card-suit__img"
    >
      Diamond.svg
    </svg>
  </div>
</div>
`;

exports[`renders correctly with Heart card 1`] = `
<div
  className="card "
>
  <div
    className="card-desc"
  >
    <span
      className="card-desc__rank card-desc__rank--red"
    >
      A
    </span>
    <svg
      className="card-desc__img"
    >
      Heart.svg
    </svg>
  </div>
  <div
    className="card-suit"
  >
    <svg
      className="card-suit__img"
    >
      Heart.svg
    </svg>
  </div>
</div>
`;

exports[`renders correctly with Spade card 1`] = `
<div
  className="card "
>
  <div
    className="card-desc"
  >
    <span
      className="card-desc__rank "
    >
      A
    </span>
    <svg
      className="card-desc__img"
    >
      Spade.svg
    </svg>
  </div>
  <div
    className="card-suit"
  >
    <svg
      className="card-suit__img"
    >
      Spade.svg
    </svg>
  </div>
</div>
`;
